### ISSUE TITLE
How to find these data from my Revit Model ? (FORGE_MODEL_URN) , (FORGE_MODEL_VIEW), (DEVICE_MODEL_ID )

### ISSUE THREAD

**Issue by Mojgan-arch** [petrbroz/forge-iot-extensions-demo#3]
**How to find these data from my Revit Model ? (FORGE_MODEL_URN) , (FORGE_MODEL_VIEW), (DEVICE_MODEL_ID )**

I need these data to use in coding. How can I get this information from my model?

1.  (FORGE_MODEL_URN) ? 
2.  (FORGE_MODEL_VIEW) ? 
3.  (DEVICE_MODEL_ID) ?

 

![jjj](https://user-images.githubusercontent.com/70357683/223854741-9291a733-bc9e-4c85-ba0f-3e6868a1cc0d.jpg)

**Comment by yiskang** [comment]
Hi @Mojgan-arch 

Please follow the [Translate a Revit File, Generating Room and Space Information tutorial](https://forge.autodesk.com/en/docs/model-derivative/v2/tutorials/prep-roominfo4viewer/) of the Model Derivative API first to translate your Revit model. When requesting the translation job, please use this job configuration instead. The `FORGE_MODEL_URN` is `BASE64_ENCODED_URN_OF_SOURCE_FILE` in task 3.

The `FORGE_MODEL_VIEW` is the `viewableID` of the Revit phase view.

Sorry, I don't see any variable called `DEVICE_MODEL_ID` in this repo.

**Comment by Mojgan-arch** [comment]
Thank you so much for your reply. 
About **(Device_Model_ID)**, In [forge-dataviz-iot-reference-app]

https://github.com/Autodesk-Forge/forge-dataviz-iot-reference-app/blob/main/server/gateways/synthetic-data/devices.json

![image](https://user-images.githubusercontent.com/70357683/205928847-8711d5e2-19d4-4484-8cf7-bfd886734a87.png)

Please guide me also, if you know about (Device_Model_ID).

**Comment by petrbroz** [comment]
This looks like a randomly generated UUID. You can generate these using different 3rd party tools, for example, https://www.uuidgenerator.net.

**Comment by yiskang** [comment]
Yeah, as Petr pointed out, it's the randomly generated UUID, and it's mainly from the `deviceId` defined in the `device-models.json`. And, see also https://github.com/petrbroz/forge-iot-extensions-demo/blob/master/public/sensormanager.js#L234 in this repo. We remove both JSON files from this repo and use a file-based JSON DB instead for that.

https://github.com/Autodesk-Forge/forge-dataviz-iot-reference-app/blob/main/server/gateways/synthetic-data/device-models.json

![enter image description here](https://i.stack.imgur.com/wt2ho.png)

**Comment by Mojgan-arch** [comment]
I got it, Thank you so much from petr broz and Yiskang.
